agents/agents-features/agents-features-opentelemetry/src/jvmMain/kotlin/ai/koog/agents/features/opentelemetry/feature/OpenTelemetry.kt

```diff
@@ -302,27 +290,20 @@ public class OpenTelemetry {
 
                 val inferenceSpan = spanProcessor.getSpanOrThrow<InferenceSpan>(inferenceSpanId)
 
-                val provider = eventContext.model.provider
-
-                // Add events to the InferenceSpan before finishing the span
-                val lastMessage = eventContext.responses.lastOrNull()
-
-                val moderationResult = eventContext.moderationResponse
-
-                val events: List<GenAIAgentEvent> = lastMessage?.let { message ->
-                    buildList {
-                        when (message) {
-                            is Message.Assistant -> add(ChoiceEvent(provider, message, config.isVerbose))
-                            else -> {}
+                if (config.useEvents) {
+                    addAfterLLMCallEvents(inferenceSpan, eventContext, config)
+                } else {
+                    eventContext.prompt.messages
+                        .filter {
+                            // TODO: check tool message here
+                            it.role == Message.Role.Assistant
                         }
-                    }
-                } ?: moderationResult?.let {
-                    buildList { add(ModerationResponseEvent(provider, it, config.isVerbose)) }
-                } ?: emptyList()
-
-                inferenceSpan.addEvents(events)
+                        .forEachIndexed { index, message ->
+                            setMessageAttributes(inferenceSpan.span, index, message, config, "completion")
+                            inferenceSpan.span.setAttribute("gen_ai.completion.$index.finish_reason", "stop")
```

Comment:
Please do not use raw attribute keys. For "finish_reason", we have `SpanAttributes.Response.FinishReason` that get an available options as a parameter of type `FinishReasonType`. I also do not see the example for `gen_ai.completion.$index.finish_reason` in OT Semantic convention. As far as I can see the "finish_reason" is only applicable as an attribute with key `gen_ai.response.finish_reason`. Could you please point me to the doc if I miss this?